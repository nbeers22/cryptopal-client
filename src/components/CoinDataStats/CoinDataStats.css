.CoinData-stats{
  background-color: #fff;
}
.CoinData-stats .stats-table{
  display: flex;
}
.CoinData-stats aside{
  flex-grow: 1;
  text-align: center;
}

@media(max-width: 767px){
  .CoinData-stats .stats-table{
    flex-wrap: wrap;
  }
  .CoinData-stats aside{
    flex: 0 0 50%;
    margin-bottom: 30px;
  }
}