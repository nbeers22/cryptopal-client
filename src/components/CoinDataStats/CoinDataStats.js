import React, { Component } from 'react'
import './CoinDataStats.css'

export default class CoinDataStats extends Component {
  state = {
    market_cap: 0,
    percent_change_1h: 0,
    percent_change_24h: 0,
    percent_change_7d: 0,
    volume_24h: 0
  }

  componentWillReceiveProps(props){
    this.setState({ ...props.prices })
  }

  currencyFormat(num){
    return num && '$' + num.toFixed(2).replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,')
  }

  percentClass(percent){
    return percent > 0 ? 'green' : 'red';
  }

  render() {
    const { 
      market_cap,
      percent_change_1h,
      percent_change_24h,
      percent_change_7d,
      volume_24h
    } = this.state

    return (
      <section className="CoinData-stats">
      <div className="container">
        <div className="stats-table">
          <aside>
            <p><strong>Market Cap</strong></p>
            { this.currencyFormat(market_cap) }
          </aside>
          <aside>
            <p><strong>Change 1hr</strong></p>
            <span className={this.percentClass(percent_change_1h)}>{ percent_change_1h.toFixed(2) + '%' }</span>
          </aside>
          <aside>
            <p><strong>Change 24hr</strong></p>
            <span className={this.percentClass(percent_change_24h)}>{ percent_change_24h.toFixed(2) + '%' }</span>
          </aside>
          <aside>
            <p><strong>Change 7d</strong></p>
            <span className={this.percentClass(percent_change_7d)}>{ percent_change_7d.toFixed(2) + '%' }</span>
          </aside>
          <aside>
            <p><strong>Volume 24hr</strong></p>
            { this.currencyFormat(volume_24h) }
          </aside>
        </div>
      </div>
    </section>
    )
  }
}
