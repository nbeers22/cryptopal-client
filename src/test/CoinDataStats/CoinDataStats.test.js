import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import CoinDataStats from '../../components/CoinDataStats/CoinDataStats.js';

describe('CoinDataStats Component',  () => {
  let wrapper;
  beforeEach( () => {
    wrapper = shallow(<CoinDataStats />)
  })
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <CoinDataStats />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders a div with className ".stats-table"', () => {
    expect(wrapper.find('.stats-table').length).toEqual(1)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('renders five columns (aside) inside .stats-table', () => {
    expect(wrapper.find('.stats-table').find('aside').length).toEqual(5)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('first column text should be "Market Cap0" when no props are set', () => {
    expect(wrapper.find('.stats-table').find('aside').at(0).text()).toEqual("Market Cap0")
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('first column text should be "Market Cap$1.00" when props are set to 1', () => {
    wrapper.setProps({ prices: { market_cap: 1 } })
    expect(wrapper.find('.stats-table').find('aside').at(0).text()).toEqual("Market Cap$1.00")
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('second column text should be "Change 1hr0" when no props are set', () => {
    expect(wrapper.find('.stats-table').find('aside').at(1).text()).toEqual("Change 1hr0.00%")
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('second column text should be "Change 1hr1.05%" when props are set to 1.05', () => {
    wrapper.setProps({ prices: { percent_change_1h: 1.05 } })
    expect(wrapper.find('.stats-table').find('aside').at(1).text()).toEqual("Change 1hr1.05%")
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('should change the class name of the percent span tags to red when props are a negative number', () => {
    wrapper.setProps({ prices: { percent_change_1h: -1.05, percent_change_24h: -2.98, percent_change_7d: -8.90 } })
    expect(wrapper.find('.stats-table').find('aside').at(1).find('span').hasClass('red')).toEqual(true)
    expect(wrapper.find('.stats-table').find('aside').at(2).find('span').hasClass('red')).toEqual(true)
    expect(wrapper.find('.stats-table').find('aside').at(3).find('span').hasClass('red')).toEqual(true)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('should change the class name of the percent span tags to green when props are a positive number', () => {
    wrapper.setProps({ prices: { percent_change_1h: 1.05, percent_change_24h: 2.98, percent_change_7d: 8.90 } })
    expect(wrapper.find('.stats-table').find('aside').at(1).find('span').hasClass('green')).toEqual(true)
    expect(wrapper.find('.stats-table').find('aside').at(2).find('span').hasClass('green')).toEqual(true)
    expect(wrapper.find('.stats-table').find('aside').at(3).find('span').hasClass('green')).toEqual(true)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
