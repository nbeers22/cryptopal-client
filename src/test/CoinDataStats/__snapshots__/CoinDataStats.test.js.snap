// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinDataStats Component first column text should be "Market Cap$1.00" when props are set to 1 1`] = `
<section
  className="CoinData-stats"
>
  <div
    className="container"
  >
    <div
      className="stats-table"
    >
      <aside>
        <p>
          <strong>
            Market Cap
          </strong>
        </p>
        $1.00
      </aside>
      <aside>
        <p>
          <strong>
            Change 1hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 24hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 7d
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Volume 24hr
          </strong>
        </p>
        0
      </aside>
    </div>
  </div>
</section>
`;

exports[`CoinDataStats Component first column text should be "Market Cap0" when no props are set 1`] = `
<section
  className="CoinData-stats"
>
  <div
    className="container"
  >
    <div
      className="stats-table"
    >
      <aside>
        <p>
          <strong>
            Market Cap
          </strong>
        </p>
        0
      </aside>
      <aside>
        <p>
          <strong>
            Change 1hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 24hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 7d
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Volume 24hr
          </strong>
        </p>
        0
      </aside>
    </div>
  </div>
</section>
`;

exports[`CoinDataStats Component renders a div with className ".stats-table" 1`] = `
<section
  className="CoinData-stats"
>
  <div
    className="container"
  >
    <div
      className="stats-table"
    >
      <aside>
        <p>
          <strong>
            Market Cap
          </strong>
        </p>
        0
      </aside>
      <aside>
        <p>
          <strong>
            Change 1hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 24hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 7d
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Volume 24hr
          </strong>
        </p>
        0
      </aside>
    </div>
  </div>
</section>
`;

exports[`CoinDataStats Component renders five columns (aside) inside .stats-table 1`] = `
<section
  className="CoinData-stats"
>
  <div
    className="container"
  >
    <div
      className="stats-table"
    >
      <aside>
        <p>
          <strong>
            Market Cap
          </strong>
        </p>
        0
      </aside>
      <aside>
        <p>
          <strong>
            Change 1hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 24hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 7d
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Volume 24hr
          </strong>
        </p>
        0
      </aside>
    </div>
  </div>
</section>
`;

exports[`CoinDataStats Component second column text should be "Change 1hr0" when no props are set 1`] = `
<section
  className="CoinData-stats"
>
  <div
    className="container"
  >
    <div
      className="stats-table"
    >
      <aside>
        <p>
          <strong>
            Market Cap
          </strong>
        </p>
        0
      </aside>
      <aside>
        <p>
          <strong>
            Change 1hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 24hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 7d
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Volume 24hr
          </strong>
        </p>
        0
      </aside>
    </div>
  </div>
</section>
`;

exports[`CoinDataStats Component second column text should be "Change 1hr1.05%" when props are set to 1.05 1`] = `
<section
  className="CoinData-stats"
>
  <div
    className="container"
  >
    <div
      className="stats-table"
    >
      <aside>
        <p>
          <strong>
            Market Cap
          </strong>
        </p>
        0
      </aside>
      <aside>
        <p>
          <strong>
            Change 1hr
          </strong>
        </p>
        <span
          className="green"
        >
          1.05%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 24hr
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 7d
          </strong>
        </p>
        <span
          className="red"
        >
          0.00%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Volume 24hr
          </strong>
        </p>
        0
      </aside>
    </div>
  </div>
</section>
`;

exports[`CoinDataStats Component should change the class name of the percent span tags to green when props are a positive number 1`] = `
<section
  className="CoinData-stats"
>
  <div
    className="container"
  >
    <div
      className="stats-table"
    >
      <aside>
        <p>
          <strong>
            Market Cap
          </strong>
        </p>
        0
      </aside>
      <aside>
        <p>
          <strong>
            Change 1hr
          </strong>
        </p>
        <span
          className="green"
        >
          1.05%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 24hr
          </strong>
        </p>
        <span
          className="green"
        >
          2.98%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 7d
          </strong>
        </p>
        <span
          className="green"
        >
          8.90%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Volume 24hr
          </strong>
        </p>
        0
      </aside>
    </div>
  </div>
</section>
`;

exports[`CoinDataStats Component should change the class name of the percent span tags to red when props are a negative number 1`] = `
<section
  className="CoinData-stats"
>
  <div
    className="container"
  >
    <div
      className="stats-table"
    >
      <aside>
        <p>
          <strong>
            Market Cap
          </strong>
        </p>
        0
      </aside>
      <aside>
        <p>
          <strong>
            Change 1hr
          </strong>
        </p>
        <span
          className="red"
        >
          -1.05%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 24hr
          </strong>
        </p>
        <span
          className="red"
        >
          -2.98%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Change 7d
          </strong>
        </p>
        <span
          className="red"
        >
          -8.90%
        </span>
      </aside>
      <aside>
        <p>
          <strong>
            Volume 24hr
          </strong>
        </p>
        0
      </aside>
    </div>
  </div>
</section>
`;
